src/compiler/watchedProgram.ts

```diff
@@ -362,7 +362,31 @@ namespace ts {
                 return !isString(hostSourceFileInfo);
             }
 
-            return host.fileExists(fileName);
+            return partialSystem.fileExists(fileName);
+        }
+
+        function directoryExists(directoryName: string) {
+            return partialSystem.directoryExists(directoryName);
+        }
+
+        function readFile(fileName: string) {

```

Comment:
For something that's only used in one place, I think it would be more readable with just an arrow function.